plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -333,6 +484,48 @@ private Boolean isModelNotDeployed(MLModelState mlModelState) {
             && !mlModelState.equals(MLModelState.PARTIALLY_DEPLOYED);
     }
 
+    private <T> Boolean isPipelineContainsModel(
+        List<T> pipelineConfigurations,
+        String candidateModelId,
+        Function<T, Map<String, Object>> getConfigFunction
+    ) {
+        for (T pipelineConfiguration : pipelineConfigurations) {
+            Map<String, Object> config = getConfigFunction.apply(pipelineConfiguration);
+            if (searchThroughConfig(config, candidateModelId, "")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private Boolean searchThroughConfig(Object searchCandidate, String candidateId, String targetModelKey) {
+        Boolean flag = false;
+        if (searchCandidate instanceof String
+            && Objects.equals(targetModelKey, PIPELINE_TARGET_MODEL_KEY)
+            && Objects.equals(candidateId, searchCandidate)) {
+            return true;
+        } else if (searchCandidate instanceof List<?>) {
+            for (Object v : (List<?>) searchCandidate) {
+                flag = flag || searchThroughConfig(v, candidateId, targetModelKey);
```

Comment:
Can you early return false?

Reply:
Already add comment and change the logic